src/System.Security.Cryptography.Xml/tests/Samples/EncryptingDecryptingAsymmetric.cs

```diff
@@ -0,0 +1,102 @@
+    // https://msdn.microsoft.com/en-us/library/ms229746(v=vs.110).aspx
+    public class EncryptingAndDecryptingAsymmetric
+    {
+        const string ExampleXmlRootElement = "example";
+        const string ExampleXml = @"<?xml version=""1.0""?>
+<example>
+<test>some text node</test>
+</example>";
+
+        private static XmlDocument LoadXmlFromString(string xml)
+        {
+            var doc = new XmlDocument();
+            doc.PreserveWhitespace = true;
+            doc.LoadXml(xml);
+            return doc;
+        }
+
+        private static void Encrypt(XmlDocument doc, string elementName, string encryptionElementID, RSA rsaKey, string keyName)
+        {
+            var elementToEncrypt = (XmlElement)doc.GetElementsByTagName(elementName)[0];
+
+            using (var sessionKey = new RijndaelManaged())
+            {
+                sessionKey.KeySize = 256;
+
+                // Encrypt the session key and add it to an EncryptedKey element.
+                var encryptedKey = new EncryptedKey()
+                {
+                    CipherData = new CipherData(EncryptedXml.EncryptKey(sessionKey.Key, rsaKey, false)),
+                    EncryptionMethod = new EncryptionMethod(EncryptedXml.XmlEncRSA15Url)
+                };
+
+                // Specify which EncryptedData
+                // uses this key. An XML document can have
+                // multiple EncryptedData elements that use
+                // different keys.
+                encryptedKey.AddReference(new DataReference()
+                {
+                    Uri = "#" + encryptionElementID
+                });
+
+                var encryptedData = new EncryptedData()
+                {
+                    Type = EncryptedXml.XmlEncElementUrl,
+                    Id = encryptionElementID,
+
+                    // Create an EncryptionMethod element so that the
+                    // receiver knows which algorithm to use for decryption.
+                    EncryptionMethod = new EncryptionMethod(EncryptedXml.XmlEncAES256Url)
+                };
+
+                encryptedData.KeyInfo.AddClause(new KeyInfoEncryptedKey(encryptedKey));
+                encryptedKey.KeyInfo.AddClause(new KeyInfoName()
+                {
+                    Value = keyName
+                });
+
+                var encryptedXml = new EncryptedXml();
+                encryptedData.CipherData.CipherValue = encryptedXml.EncryptData(elementToEncrypt, sessionKey, false);
+
+                EncryptedXml.ReplaceElement(elementToEncrypt, encryptedData, false);
+            }
+        }
+
+        public static void Decrypt(XmlDocument doc, RSA rsaKey, string keyName)
+        {
+            var encrypted = new EncryptedXml(doc);
+            encrypted.AddKeyNameMapping(keyName, rsaKey);
+            encrypted.DecryptDocument();
+        }
+
+        [Fact]
+        public void AsymmetricEncryptionRoundtrip()
+        {
+            using (var key = RSA.Create())
```

Comment:
I don't know of anything that gets leaked by creating an RSA key on any OS.

In the dev/apple_crypto branch RSA keygen was disabled (until yesterday) on OSX because the keygen method would save the key into the keychain... but that's now fixed, and hasn't ever been true in master.